src/createSatchel.ts

```diff
@@ -0,0 +1,282 @@
+import { observable, ObservableMap, transaction, action as mobxAction, IAction } from 'mobx';
+import {
+    getPrivateActionId,
+    getPrivateActionType,
+    setActionType,
+    setPrivateActionId,
+    getPrivateSubscriberRegistered,
+    setPrivateSubscriberRegistered,
+} from './privatePropertyUtils';
+import ActionMessage from './interfaces/ActionMessage';
+import Middleware from './interfaces/Middleware';
+import DispatchFunction from './interfaces/DispatchFunction';
+import SubscriberFunction from './interfaces/SubscriberFunction';
+import ActionCreator from './interfaces/ActionCreator';
+import { Mutator } from './interfaces/Mutator';
+import { Subscriber } from './interfaces/Subscriber';
+
+type SatchelState = {
+    __rootStore: ObservableMap<any>;
+    __nextActionId: number;
+    __subscriptions: { [key: string]: SubscriberFunction<ActionMessage, void>[] };
+    __currentMutator: string | null;
+};
+
+export type SatchelInstance = {
```

Comment:
Nitpick: I'd prefer to call this type just `Satchel` ...unless there's some need to distinguish an instance of a satchel from something else? (I haven't made it through all the code yet, so not sure.)